npm/pkg/dataplane/dataplane.go

```diff
@@ -0,0 +1,134 @@
+package dataplane
+
+import (
+	"runtime"
+
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane/ipsets"
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane/policies"
+)
+
+type DataPlane struct {
+	DataPlaneInterface
+	policyMgr policies.PolicyManager
+	ipsetMgr  ipsets.IPSetManager
+	OsType    OsType
+	networkID string
+	// key is PodKey
+	endpointCache map[string]interface{}
+}
+
+func NewDataPlane() *DataPlane {
+	return &DataPlane{
+		OsType:    detectOsType(),
+		policyMgr: policies.NewPolicyManager(),
+		ipsetMgr:  ipsets.NewIPSetManager(string(detectOsType())),
+	}
+}
+
+type OsType string
+
+const (
+	Windows OsType = "windows"
+	Linux   OsType = "linux"
+)
+
+type DataPlaneInterface interface {
+	NewDataplane() (*DataPlane, error)
```

Comment:
would this not be in the interface because it's a function instead of a method?

Reply:
You are right, will correct that

Reply:
@vakalapa that you accidentally included the `New` in the interface kinda points at a problem with this interface: it's very big but also very specific. This hasn't _really_ decoupled anything, you've just moved where the coupling happens.
I noticed that you have naturally grouped the methods in the interface, which might indicate that those groups are the true interfaces. You could break this up in to smaller and easier to swap out interfaces and more well isolated implementations.

Reply:
Agreed, there is not strict reason to use this interface apart from being a easily way to discover dataplane APIs for some body working on the control plane side. I will remove it as soon as this PR hits some maturity.